editor.planx.uk/src/@planx/components/MultipleFileUpload/Modal.tsx

```diff
@@ -103,28 +110,60 @@ const SelectMultiple = (props: SelectMultipleProps) => {
     );
   };
 
+  const updateFileListWithTags = () => {
+    const updatedFileList: FileList = merge(fileList);
+    const categories = Object.keys(updatedFileList) as Array<
+      keyof typeof updatedFileList
+    >;
+
+    tags.forEach((tag) => {
+      categories.forEach((category) => {
+        const updatedUserFileIndex = updatedFileList[category].findIndex(
+          (fileType) => fileType.name === tag
+        );
+        if (updatedUserFileIndex > -1) {
+          const updatedFileType =
+            updatedFileList[category][updatedUserFileIndex];
+          if (updatedFileType.slots) {
+            updatedFileList[category][updatedUserFileIndex].slots?.push(
+              uploadedFile
+            );
+          } else {
+            updatedFileList[category][updatedUserFileIndex] = {
+              ...updatedFileList[category][updatedUserFileIndex],
+              slots: [uploadedFile],
+            };
+          }
+        }
+      });
+    });
+
+    setFileList(updatedFileList);
+  };
+
   return (
     <FormControl
-      key={`form-${name}`}
+      key={`form-${uploadedFile.id}`}
       sx={{ display: "flex", flexDirection: "column" }}
     >
       <InputLabel
-        id={`select-mutliple-file-tags-label-${name}`}
+        id={`select-mutliple-file-tags-label-${uploadedFile.id}`}
         sx={{ top: "15%", color: (theme) => theme.palette.text.primary }}
       >
         What does this file show?
       </InputLabel>
       <Select
-        key={`select-${name}`}
-        id={`select-multiple-file-tags-${name}`}
-        labelId={`select-multiple-file-tags-label-${name}`}
+        key={`select-${uploadedFile.id}`}
+        id={`select-multiple-file-tags-${uploadedFile.id}`}
+        labelId={`select-multiple-file-tags-label-${uploadedFile.id}`}
         variant="standard"
         multiple
         value={tags}
         onChange={handleChange}
+        onClose={updateFileListWithTags}
```

Comment:
I think we have options of where to call this function from! I've chosen here in this first iteration because it means it will update state a bit less frequently, but it works just the same if run like this: 
```jsx
useEffect(() => {
  updateFileListWithTags();
}, [tags])
```

Reply:
Cracked the logic for removing tags using `usePrevious(tags)`, therefore this call does actually need to happen within a `useEffect` hook listening for each change to `tags`, rather than `onClose` which captures a few tags changes at once. 

Disregard initial comment!